thetis/solver2d.py

```diff
@@ -263,11 +263,12 @@ def create_function_spaces(self):
                 'Spherical mesh requires \'rt-dg\' or \'bdm-dg\' ' \
                 'element family.'
         # ----- function spaces: elev in H, uv in U, mixed is W
-        self.function_spaces.P0_2d = get_functionspace(self.mesh2d, 'DG', 0, name='P0_2d')
+        DG = 'DG' if self.mesh2d.ufl_cell().cellname() == 'triangle' else 'DQ'
```

Comment:
Thanks for fixing this. I always get confused here though. Is there a difference between DQ and dP on quads? Which is the one we want?

Reply:
Good point. I just put DQ because it was what Colin put in the original demo and also what Firedrake automatically changed DG to.

Reply:
There is a difference because I think DQ1 has four DoFs on a quad and dPc1 has three.

Reply:
OK then I think we indeed want DQ1